morphia/src/main/java/org/mongodb/morphia/query/FieldEndImpl.java

```diff
@@ -142,7 +142,13 @@ public T doesNotHaveThisElement(final Object val) {
     @Override
     public T hasThisElement(final Object val) {
         Assert.parametersNotNull("val", val);
-        return addCriteria(FilterOperator.ELEMENT_MATCH, val, false);
+        return addCriteria(FilterOperator.ELEMENT_MATCH, val, not);
```

Comment:
Is that bug reported?  If not, it should be so we can track that it's getting fixed.  And if it's not already, the bug fix should be done in a separate commit from the new feature work for elemMatch